ros2cli/ros2cli/daemon/__init__.py

```diff
@@ -92,6 +96,65 @@ def shutdown_handler():
             server.server_close()
 
 
+class _DirectNode:
+    """A direct node, that resets itself when a network interface changes."""
+
+    def __init__(self, args):
+        self.args = args
+        # TODO(ivanpauno): A race condition is possible here, since it isn't possible to know
+        # exactly which interfaces were available at node creation.
+        self.node = DirectNode(args)
+        self.addresses_at_start = self.interfaces_ip_addresses()
+
+    def __enter__(self):
+        self.node.__enter__()
+        return self
+
+    def __getattr__(self, name):
+        attr = self.node.__getattr__(name)
```

Comment:
@ivanpauno nit: consider 
```suggestion
        attr = getattr(self.node, name)
```